packages/block-editor/src/components/block-list/index.native.js

```diff
@@ -321,6 +321,13 @@ export class BlockList extends Component {
 			gridProperties,
 		} = this.props;
 		const { blockWidth } = this.state;
+
+		// blockClientIds are only required for Grid blocks in the BlockListItem component.
+		// However passing the blockClientIds as a top level prop results in unneeded re-renders.
+		const gridItemProps = gridProperties && {
+			props: gridProperties,
+			items: blockClientIds,
+		};
```

Comment:
> I think one approach for this would be to flatten the grid properties
I had considered something along these lines around pulling out the `numColumns` prop but since you noticed other re-renders I'm on board for refining the grid item props in this PR 👍

Reply:
Updated in 584e75c43b7377066fa1d3c0c46265d54333f302

Reply:
I'd like to note that I still encountered re-renders caused by `gridItemProps` in the `BlockListBlock` component using the changes from https://github.com/WordPress/gutenberg/commit/584e75c43b7377066fa1d3c0c46265d54333f302.

<img width="1144" alt="Screenshot 2022-05-03 at 13 44 07" src="https://user-images.githubusercontent.com/14905380/166446997-12373c18-8c5f-4028-ac45-308d641fbeb4.png">

I think the problem is related to the fact that we create a new `gridItemProps` object on every render ([reference](https://github.com/WordPress/gutenberg/blob/584e75c43b7377066fa1d3c0c46265d54333f302/packages/block-editor/src/components/block-list/index.native.js#L328-L332)), which produces the extra re-renders in the`BlockListBlock` component as the `gridItemProps` prop is passing a new instance.

Most likely this issue could be prevented by either flattening the grid props, as I mentioned in [this comment](https://github.com/WordPress/gutenberg/pull/40705#discussion_r861965340), or memoizing the `gridItemProps` so it doesn't create a new instance on every render, wdyt?

Reply:
Ok updated in b2a49fc95e6f78894c49337cfc34d90f76af8f3b

I'm not seeing any `BlockListBlock` re-renders.